lib/services/matching_service.rb

```diff
@@ -0,0 +1,41 @@
+module Matching
+  def self.matches(game)
+    user_code_clone = game.user_code.clone
+    matches = game.secret_code.map do |number|
+      user_code_clone.find do |user_number|
+        user_code_clone.delete_at(user_code_clone.index(user_number)) if user_number == number
+      end
+    end
+    matches.compact! || matches
+  end
+
+  def self.create_response(game)
+    @pluses = ''
+    @minuses = ''
+    @spaces = ''
+    check_the_code(game)
+    "#{@pluses}#{@minuses}#{@spaces}"
+  end
+
+  def self.check_the_code(game)
```

Comment:
It is very difficult... I think you can simplify this. First of all find all exact matches and then all the rest matches.